### PR TITLE
Add whatwg/storage configs

### PR DESCRIPTION
**Pull request by saschanaz**
Corresponds to https://github.com/whatwg/storage/pull/152.

### CODE REVIEW

**Comment by saschanaz** [comment]
Hmm, looking at https://github.com/whatwg/storage/actions/runs/3531622937/jobs/5924996674 we seemingly need `pre_build_step` instead of the post one. I guess for now we should just checkin the SVG file, what do you think? @domenic

**Comment by saschanaz** [comment]
(Can we fix https://resources.whatwg.org/build/deploy.sh instead?)

**Comment by domenic** [comment]
Yeah, maybe we should just reverse the order of `copy_extra_files` and `run_post_build_step`? The one to check would be whatwg/encoding; does that depend on the current ordering, or can be it be made to work with the opposite order?

**Comment by saschanaz** [comment]
Not even sure how is it working right now? It generates files after copy?

**Comment by domenic** [comment]
Yep, that's how it's working right now:

```bash
curlbikeshed "$INPUT_FILE" \
             "$WEB_ROOT/index.html" \
             -F md-Text-Macro="COMMIT-SHA $SHA"
copy_extra_files "$WEB_ROOT"
run_post_build_step "$WEB_ROOT"
echo "Living standard output to $WEB_ROOT"
```

**Comment by saschanaz** [comment]
Oh I mean, not sure how Encoding is happy with that... Does it really getting the generated files?

**Comment by annevk** [comment]
@saschanaz Encoding just generates a bunch of additional files. Those files are themselves not inputs to anything else.

File: factory.json [hunk 1]
```diff
@@ -32,6 +32,12 @@
   "infra": {
     "not_these_templates": ["PULL_REQUEST_TEMPLATE.md"]
   },
+  "storage": {
+    ".gitignore": ["/node_modules/"],
+    "build_with_node": true,
+    "extra_files": "assets/*",
+    "post_build_step": "npm run build-diagram"
```
Comment (annevk) [inline factory.json:39]: I don't think we need this? We only need to run this when we change the diagram, which is hopefully not too often.

**Comment by saschanaz** [comment]
But they should be served to be referenced via anchor elements, no?

File: factory.json (hunk 1 above)
Comment (saschanaz) [reply factory.json:39]: Yeah, we are blocked anyway, let's just checkin the SVG for now.

File: factory.json [hunk 2]
```diff
@@ -32,6 +32,11 @@
   "infra": {
     "not_these_templates": ["PULL_REQUEST_TEMPLATE.md"]
   },
+  "storage": {
+    ".gitignore": ["/node_modules/"],
+    "build_with_node": true,
```
Comment (annevk) [inline factory.json:37]:
```suggestion
```

**Review by annevk — COMMENTED** [review]
Do you mean that the MIME type is not set correctly currently? That's strange, but we should be able to sort that on https://github.com/whatwg/misc-server.

**Comment by saschanaz** [comment]
No, I mean the resulting files from visualize.py, which is called as a post build step. But that script interestingly directly writes to the web root. Got it 🙂

File: factory.json [hunk 3]
```diff
@@ -32,6 +32,10 @@
   "infra": {
     "not_these_templates": ["PULL_REQUEST_TEMPLATE.md"]
   },
+  "storage": {
+    ".gitignore": ["/node_modules/"],
```
Comment (annevk) [inline factory.json:36]: So I think that for now we want to ignore the lockfile as well given that we will generate locally and compare the output on git.

Comment (saschanaz) [reply factory.json:36]: Yeah, let's do that.

```suggestion
    ".gitignore": ["/node_modules/", "/package-lock.json"],
```

**Review by annevk — APPROVED** [review]
Thanks. This looks good to me. Hopefully @domenic can give this a final look next week as well as the other PR.

**Review by domenic — APPROVED** [review]

**Pull request merged**